plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -256,7 +256,7 @@ public ConnectorPageSource createPageSource(
                 requiredColumns,
                 dataPageSource.get(),
                 projectionsAdapter,
-                deleteFilter);
+                Optional.of(deleteFilter).filter(filter -> filter.hasPosDeletes() || filter.hasEqDeletes()));
```

Comment:
Why is it needed?

Reply:
The data Page is always rewritten in the IcebergPageSource right now, this makes it conditional on if there are actually any deletes to account for.

Reply:
sorry for not being clear. I understand what the change is for.
wonder why `Optional#filter` is needed here.
if "deleteFilter" is not needed, why was it passed until here? did the split source attach some useless information to the split?

Reply:
Oh, I see. We could also make this conditional on `split.getDeletes().isEmpty()`  if that's clearer? They're equivalent

Reply:
> why was it passed until here?

The DeleteFilter is constructed on the line right above this, the only thing passed around until here are some empty lists. `split.getDeletes()` should be empty, as should `deleteFilterRequiredSchema`